### PR TITLE
Update TTS recipes to use config files

### PR DESCRIPTION
**Pull request by kan-bayashi**
This is the one of commits in #752.

### CODE REVIEW

**Comment by sw005320** [comment]
Thanks. My only concern is the change of the option name conventions (from `_` to `-`). This may affect some of them, but it would be OK as a major v.0.4.0 revision (one option is to keep the original option names).

**Comment by kan-bayashi** [comment]
Ready to merge.

**Comment by kan-bayashi** [comment]
I will check with `egs/an4/tts1`.

**Comment by kan-bayashi** [comment]
OK.

File: egs/an4/tts1/run.sh [hunk 1]
```diff
@@ -182,38 +144,7 @@ if [ ${stage} -le 2 ] && [ ${stop_stage} -ge 2 ]; then
 fi
 
 if [ -z ${tag} ];then
```
Comment (takenori-y) [inline egs/an4/tts1/run.sh]: Could you add a space after semicolon (in all run.sh files)?

File: egs/ljspeech/tts2/run.sh [hunk 2]
```diff
@@ -343,7 +219,6 @@ if [ ${stage} -le 6 ] && [ ${stop_stage} -ge 6 ];then
             --n_fft ${n_fft} \
             --n_shift ${n_shift} \
             --win_length "${win_length}" \
-            --iters ${griffin_lim_iters} \
```
Comment (takenori-y) [inline egs/ljspeech/tts2/run.sh]: Is it ok to remove `--iters` option? This is inconsistent with `egs/ljspeech/tts1/run.sh`.

File: egs/m_ailabs/tts1/conf/train_pytorch_tacotron2.yaml [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+adim: 128
+aconv-chans: 32
+aconv-filts: 15      # resulting in filter-size = aconv-filts * 2 + 1
+cumulate-att-w: true # whether to cumulate attetion weight
+use-batch-norm: true # whether to use batch normalization in conv layer
+use-concate: true    # whether to concatenate encoder embedding with decoder lstm outputs
+use-residual: false  # whether to use residual connection in encoder convolution
+use-masking: true    # whether to mask the padded part in loss calculation
+bce-pos-weight: 1.0  # weight for positive samples of stop token in cross-entropy calculation
+reduction-factor: 2
```
Comment (takenori-y) [inline egs/m_ailabs/tts1/conf/train_pytorch_tacotron2.yaml]: Check the values of `atype` and `reduction-factor`.

File: espnet/bin/tts_train.py [hunk 4]
```diff
@@ -49,19 +49,24 @@ def main(cmd_args):
                         help='Process only N minibatches (for debug)')
     parser.add_argument('--verbose', '-V', default=0, type=int,
                         help='Verbose option')
-    parser.add_argument('--tensorboard-dir', default=None, type=str, nargs='?', help="Tensorboard log dir path")
+    parser.add_argument('--tensorboard-dir', default=None, type=str, nargs='?',
+                        help="Tensorboard log dir path")
```
Comment (takenori-y) [inline espnet/bin/tts_train.py]: dir path -> directory path

File: egs/ljspeech/tts2/run.sh (hunk 2 above)
Comment (kan-bayashi) [reply egs/ljspeech/tts2/run.sh]: In the case of `tts2` recipes, we do not perform griffin lim so it is not needed.

Comment (takenori-y) [reply egs/ljspeech/tts2/run.sh]: I see.

File: egs/an4/tts1/conf/train_pytorch_tacotron2.yaml [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+maxlen-in: 150     # if input length  > maxlen-in, batchsize is reduced (if use "shuffle", not effect)
+maxlen-out: 400    # if output length > maxlen-out, batchsize is reduced (if use "shuffle", not effect)
+
+# optimization related
+lr: 1e-3
+eps: 1e-6
+weight-decay: 0.0
+dropout-rate: 0.5
+zoneout-rate: 0.1
+epochs: 50
```
[the remark this answers is not in the document]
Comment (kan-bayashi) [reply egs/an4/tts1/conf/train_pytorch_tacotron2.yaml]: `an4` recipe is the kind of operation check scripts.
I think this is enough.

File: espnet/bin/tts_train.py (hunk 4 above)
Comment (kan-bayashi) [reply espnet/bin/tts_train.py]: Fixed.

File: egs/an4/tts1/run.sh (hunk 1 above)
Comment (kan-bayashi) [reply egs/an4/tts1/run.sh]: Fixed.

File: egs/ljspeech/tts1/conf/train_pytorch_tacotron2.yaml [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+adim: 128
+aconv-chans: 32
+aconv-filts: 15      # resulting in filter-size = aconv-filts * 2 + 1
+cumulate-att-w: true # whether to cumulate attetion weight
+use-batch-norm: true # whether to use batch normalization in conv layer
+use-concate: true    # whether to concatenate encoder embedding with decoder lstm outputs
+use-residual: false  # whether to use residual connection in encoder convolution
+use-masking: true    # whether to mask the padded part in loss calculation
+bce-pos-weight: 1.0  # weight for positive samples of stop token in cross-entropy calculation
+reduction-factor: 2
```
[the remark this answers is not in the document]
Comment (kan-bayashi) [reply egs/ljspeech/tts1/conf/train_pytorch_tacotron2.yaml]: Actually I’m not sure which is better.
So I added two configs and then make symlink as the same manner in ASR.

File: egs/m_ailabs/tts1/conf/train_pytorch_tacotron2.yaml (hunk 3 above)
Comment (kan-bayashi) [reply egs/m_ailabs/tts1/conf/train_pytorch_tacotron2.yaml]: Same as the above.

**Comment by kan-bayashi** [comment]
Thank you for your review, @takenori-y !
I reflected your comments.

**Comment by kan-bayashi** [comment]
Fixed wrong URL #751

File: egs/ljspeech/tts2/run.sh (hunk 2 above)
Comment (kan-bayashi) [reply egs/ljspeech/tts2/run.sh]: Sorry, this is worng.
It uses Grrifin-Lim. I confused with another method...

**Pull request merged**